src/validator.ts

```diff
@@ -188,49 +189,268 @@ export const possiblyReturningBlock = warning<Method>(node => {
+})
+
+export const shouldNotDefineUnnecesaryIf = error<If>(node =>
+  notEmpty(node.elseBody.sentences) || !node.condition.is('Literal') || node.condition.value !== true
+)
+
+export const shouldNotDefineEmptyDescribe = warning<Describe>(node =>
+  notEmpty(node.tests())
+)
+
+export const shouldHaveNonEmptyName = warning<Describe | Test>(node =>
+  (node.name ?? '').replaceAll('"', '').trim() !== ''
+)
+
+export const shouldNotMarkMoreThanOneOnlyTest = warning<Test>(node =>
+  !node.isOnly || node.parent().children().filter(element => element.is('Test') && element.isOnly).length <= 1
+)
+
+export const shouldNotDefineNativeMethodsOnUnnamedSingleton = error<Method>(node => {
+  const parent = node.parent()
+  return !node.isNative() || !parent.is('Singleton') || !!parent.name
+})
+
 // ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
 // HELPER FUNCTIONS
 // ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
 
 const allParents = (module: Module) =>
   module.supertypes.map(supertype => supertype.reference.target()).flatMap(supertype => supertype?.hierarchy() ?? [])
 
+const getReferencedModule = (parent: Node): Module | undefined => {
+  if (parent.kind === 'ParameterizedType') return parent.reference.target()
+  if (parent.kind === 'New') return parent.instantiated.target()
+  return undefined
+}
+
+const uninitializedValue = (value: Expression | undefined) => value && value.is('Literal') && !value.value && value.isSynthetic()
+
+const getUninitializedAttributes = (node: Module, initializers: string[] = []): string[] => {
+  const uninitializedAttributes: string[] = []
+  node.defaultFieldValues()?.forEach(
+    (value, field) => {
+      if (uninitializedValue(value) && !initializers.includes(field.name)) {
+        uninitializedAttributes.push(field.name)
+      }
+    })
+  return uninitializedAttributes
+}
+
+const isBooleanLiteral = (node: Expression, value: boolean) => node.is('Literal') && node.value === value
+
+const targetSupertypes = (node: Class | Singleton) => node.supertypes.map(_ => _?.reference.target())
+
+const superclassMethod = (node: Method) => node.parent().lookupMethod(node.name, node.parameters.length, { lookupStartFQN: node.parent().fullyQualifiedName(), allowAbstractMethods: true })
+
+const finishesFlow = (sentence: Sentence, node: Node): boolean => {
+  const parent = node.parent()
+  const lastLineOnMethod = parent.is('Body') ? last(parent.sentences) : undefined
+  const returnCondition = (sentence.is('Return') && lastLineOnMethod !== node && lastLineOnMethod?.is('Return') || lastLineOnMethod?.is('Throw')) ?? false
+  return sentence.is('Throw') || sentence.is('Send') || sentence.is('Assignment') || sentence.is('If') || returnCondition
+}
+
+const isGlobal = (node: Variable) => node.parent().is('Package')
+
+const getVariableContainer = (node: Node): Method | Test => {
```

Comment:
Esto no podría hacerse con `node.ancestors().find(is('Entity')) ?

Reply:
no tenía esa función... pero me viene al pelo

Reply:
Igual no es equivalente preguntar por `is('Entity')`, pero sí voy a tomar esa definición que mola más